File: batou-rules/testdata/fixtures/bench/zig/vulnerable/a02_weak_crypto.zig
// Source: CWE-327 - Use of weak Md5 hash in Zig
// Expected: BATOU-ZIG
// OWASP: A02:2021 - Cryptographic Failures

const std = @import("std");

pub fn hashPassword(password: []const u8) [32]u8 {
    var hash: [std.crypto.hash.Md5.digest_length]u8 = undefined;
    std.crypto.hash.Md5.hash(password, &hash, .{});
    return std.fmt.bytesToHex(hash, .lower);
}

pub fn main() !void {
    const result = hashPassword("admin123");
    const stdout = std.io.getStdOut().writer();
    try stdout.print("MD5: {s}\n", .{&result});
}

File: batou-rules/testdata/fixtures/bench/csharp/vulnerable/a02_weak_crypto.cs
// Source: CWE-327 - Use of weak MD5 hash in C#
// Expected: BATOU-CRY
// OWASP: A02:2021 - Cryptographic Failures

using System;
using System.Security.Cryptography;
using System.Text;

namespace App
{
    class CryptoHelper
    {
        public string HashPassword(string password)
        {
            using (var md5 = MD5.Create())
            {
                var bytes = Encoding.UTF8.GetBytes(password);
                var hash = md5.ComputeHash(bytes);
                return BitConverter.ToString(hash).Replace("-", "");
            }
        }

        static void Main()
        {
            var helper = new CryptoHelper();
            Console.WriteLine(helper.HashPassword("admin123"));
        }
    }
}
